src/Compiler/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTypeVisitor.cs

```diff
@@ -26,6 +26,14 @@ internal class ViewComponentTypeVisitor : SymbolVisitor
         _results = results;
     }
 
+    public override void VisitAssembly(IAssemblySymbol symbol)
+    {
+        if (ViewComponentTagHelperDescriptorProvider.IsTagHelperAssembly(symbol))
```

Comment:
Let's file a bug on us to revisit these checks and remove where appropriate, but for this servicing fix we'll take the minimum change.

Reply:
https://github.com/dotnet/razor/issues/8336